### ISSUE TITLE
Method Konnco\FilamentImport\Actions\ImportAction::table does not exist.

### ISSUE THREAD

**Issue by 4inka** [konnco/filament-import#30]
I have a resource that has a relation manager associated, and I would like to be able to import data to the relation but when I add the import action it gives me an error saying `Method Konnco\FilamentImport\Actions\ImportAction::table does not exist.`.

After investigating I noticed that the `ImportAction` extends `Filament\Pages\Actions\Action` and to be able to add an action to a tables `headerActions` param it has to extend `Filament\Tables\Actions\Action`. Could it be possible to create another `ImportAction` that extends the tables class so it can be usable inside relations and other places that can only be extended from `Filament\Tables\Actions\Action`?

**Comment by arjendejong12** [comment]
This would be so good to have, as I'm not using Filament resources, only forms and tables. If I change the mentioned `extend` to `Filament\Tables\Actions\Action`, the import button shows up, else I get the same error @4inka has.

**Comment by TonyPartridgeR** [comment]
This is really needed and extends the functionality massively to all filamentphp.

**Comment by tomnewman86** [comment]
Was there a reliable workaround to this issue? 

I too am getting the `Method Konnco\FilamentImport\Actions\ImportAction::table does not exist` when trying to use this plugin with filament/tables only, in the getTableHeaderActions() function

Thanks

**Comment by tonypartridge** [comment]
Not at present, it's only supported on the Resource level.

**Comment by pathros** [comment]
How to implement the import inside a relation manager ??? Because I got the following error (when implemented inside the `->headerActions([])` method of the RelationManager)

```php
public static function table(Table $table): Table{
    return $table
        ->headerActions([
             ImportAction::make()
    ,]);
```

> Method Konnco\FilamentImport\Actions\ImportAction::table does not exist.

**Comment by tonypartridge** [comment]
It needs to add individual support for table actions I believe whereas it's built for resources at the moment

**Comment by ayatamine** [comment]
hi, is there any new updates for this, i have the same issue. thanks

**Comment by vy-shmal** [comment]
I also am using ```filament table``` without the admin panel. 
Had the same issue with the ```Method Konnco\FilamentImport\Actions\ImportAction::table does not exist```  when used in 
```PHP
protected function getTableHeaderActions(): array
{
    return []
}
```

I don't know if its the best practice but to make it work I made an overwrite of the class with my custom class, changing only the extend to ```\Filament\Tables\Actions\Action``` as @4inka suggested and adding the overwrite  inside `\App\Providers\AppServiceProvider`
```PHP
   public function register(): void
    {
        $loader = AliasLoader::getInstance();
        $loader->alias('Konnco\FilamentImport\Actions\ImportAction', 'App\Rewrites\Filament\ImportAction');


    }
```
Hope soon we will have a better solution.

**Comment by sprklinginfo** [comment]
+1, any update?

**Comment by jhoanborges** [comment]
+1

**Comment by momostafa** [comment]
much needed +1

**Comment by tonypartridge** [comment]
> much needed +1

this is possible with the native Filament Importer

**Comment by momostafa** [comment]
> > much needed +1
> 
> this is possible with the native Filament Importer

Yes but I find this plugin more useful than the native thanks to the developer and the contributors
